File: 04-Arrays/1D-Arrays/IntersectionOfTwoArray.cpp
#include <iostream>
#include <vector>
using namespace std;

// i/p
// 6
// 1 2 3 4 6 8
// 4 
// 3 4 9 10

// o/p
// 3 4

vector<int> InterSection(vector<int> arr,vector<int> brr){

    vector<int> ans;

    for(int i = 0 ; i < arr.size() ;i++){
        int k = arr[i];
        for(int j = 0 ; j < brr.size(); j++){
            if(k == brr[j]){

                //marking the value
                brr[j] = -1;
                ans.push_back(k);

            }
        }
    }



    return ans;
}

int main() {

    int n1;
    cin>>n1;

    vector<int> arr(n1);

    for(int i = 0 ; i< n1 ; i++){
        cin>>arr[i];
    }

    int n2;
    cin>>n2;
    vector<int> brr(n2);
    for(int i = 0 ; i< n2 ; i++){
        cin>>brr[i];
    }

    vector<int> ans = InterSection(arr,brr);
    for(int i = 0 ; i< ans.size() ; i++){
        cout<<ans[i]<<" ";
    }
 

    return 0;
}

File: 05-Array_List-InPython/1D Array in python/IntersectionOfTwoArray.py


# Input : 
# lst1 = [15, 9, 10, 56, 23, 78, 5, 4, 9]
# lst2 = [9, 4, 5, 36, 47, 26, 10, 45, 87]
# Output :
# [9, 10, 4, 5]


def InterSection(arr,brr):

    # //set using due to avoid unique values 

    lst1 = set(arr)
    # print(lst1)
    lst2 = set(brr)
    # print(lst2)
    
    ans = []

    for i in lst1:
        if(i in lst2):
            
            ans.append(i)
        
    
    return ans



lst1 = [15, 9, 10, 56, 23, 78, 5, 4, 9]
lst2 = [9, 4, 5, 36, 47, 26, 10, 45, 87]

res = InterSection(lst1,lst2)
print(res)